mozilla-services/syncstorage-rs
# feat: User migration scripts

Issue: #286

## Description
Several scripts to assist in user migration from MySQL to spanner. Consult the `README.md` file for details on use and configuration.

## Testing

1. Set up a [local syncserver](https://mozilla-services.readthedocs.io/en/latest/howtos/run-sync-1.5.html), connect, and store some BSO data.
2. Configure the `move_dsns.lst` config file to point to the correct mysql and spanner instances. (e.g. 
```
mysql://test:test@localhost/syncstorage
spanner://projects/sync-spanner-dev-225401/instances/spanner-test/databases/sync_schema3
```
3. Store the known local syncserver userID into `move_users.lst` (e.g. `1`)
4. After running setup (`python3 -m venv venv && . venv/bin/activate && pip install -r requirements.txt`) Run the `python3 migrate_user.py` script.
5. user data record should be copied from mysql to spanner, Note the 
`Processsing... # -> ###:###` 
which displays the original mysql `userid` and the spanner `fxa_uid:fxa_kid`

## Issue(s)

Issue #286, https://github.com/mozilla-services/services-engineering/issues/18
